st2api/st2api/controllers/v1/keyvalue.py

```diff
@@ -55,26 +59,37 @@ def __init__(self):
         self._coordinator = coordination.get_coordinator()
         self.get_one_db_method = self._get_by_name
 
-    @jsexpose(arg_types=[str, str, bool])
-    def get_one(self, name, scope=SYSTEM_SCOPE, decrypt=False):
+    @jsexpose(arg_types=[str, str, str, bool])
+    def get_one(self, name, scope=SYSTEM_SCOPE, user=None, decrypt=False):
         """
             List key by name.
 
             Handle:
                 GET /keys/key1
         """
         self._validate_scope(scope=scope)
-        key_ref = get_key_reference(scope=scope, name=name, user=get_requester())
+
+        if user:
+            # Providing a user implies a user scope
+            scope = USER_SCOPE
+
+        requester_user = get_requester()
+        user = user or requester_user
+        is_admin = request_user_is_admin(request=pecan.request)
+
+        # User needs to be either admin or requesting item for itself
+        self._validate_decrypt_query_parameter(decrypt=decrypt, scope=scope, is_admin=is_admin)
+
+        # Validate that the authenticated user is admin if user query param is provided
+        self._validate_user_query_parameter(user=user, is_admin=is_admin)
+
+        key_ref = get_key_reference(scope=scope, name=name, user=user)
         from_model_kwargs = {'mask_secrets': not decrypt}
-        try:
-            kvp_api = self._get_one_by_scope_and_name(
-                name=key_ref,
-                scope=scope,
-                from_model_kwargs=from_model_kwargs
-            )
-        except StackStormDBObjectNotFoundError as e:
-            abort(http_client.NOT_FOUND, e.message)
-            return
+        kvp_api = self._get_one_by_scope_and_name(
```

Comment:
I don't know why the except clause was removed without a corresponding change in _get_one_by_scope_and_name? Am I missing something?